<a href="https://colab.research.google.com/github/medhu07/flyrankai/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/medhu07/flyrankai/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## My Lane

**Chosen Lane:** Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because it improves a clear business decision: deciding which content pages should be reviewed first. Rather than automatically deciding which pages should be updated, the goal is to create a ranked review queue that helps content strategists prioritize their limited time. This lane fits naturally as a ranking problem because pages can be ordered by priority instead of being classified as simply good or bad. The starter dataset also provides observable search and engagement signals that can be combined to support this decision throughout the internship.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## The Question

### Search Question

Which content pages should be reviewed first for refresh based on their observable search and engagement signals?

### Decision

Prioritize which pages deserve editorial review before others.

### Unit of Analysis

One content page.

### Output

A ranked list of content pages with a refresh priority score and supporting reason codes.

### Action

A content strategist reviews the highest-ranked pages and decides whether to refresh, expand, monitor, or leave the content unchanged.

### Cost of a Wrong Recommendation

A false positive wastes editorial effort by recommending pages that do not require attention. A false negative may cause genuinely declining pages to be overlooked, potentially missing opportunities to improve search performance.

### Why Data / ML Helps

Many observable signals such as impressions, CTR, engagement, freshness, and content characteristics interact in ways that are difficult to capture with simple rules alone. A ranking model can combine these signals to better prioritize review candidates while still supporting human decision-making.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
!pwd
!ls

/content
sample_data


In [3]:
!git clone https://github.com/medhu07/flyrankai.git

Cloning into 'flyrankai'...
remote: Enumerating objects: 123, done.
remote: Counting objects: 100% (123/123), done.
remote: Compressing objects: 100% (79/79), done.
remote: Total 123 (delta 38), reused 97 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (123/123), 1.84 MiB | 13.88 MiB/s, done.
Resolving deltas: 100% (38/38), done.


In [5]:
%cd flyrankai

/content/flyrankai


In [6]:
!ls

AGENTS.md  DATA_USE.md	LICENSE    README.md	     SETUP.md	 work
CLAUDE.md  docs		notebooks  requirements.txt  skills
data	   GUIDE.md	outputs    scripts	     submission


In [9]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [8]:
print("Dataset Shape:", df.shape)

print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nImpressions Statistics:")
print(df["impressions_90d"].describe())

print("\nAverage CTR (%):", round(df["ctr"].mean(), 2))

Dataset Shape: (30000, 44)

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Impressions Statistics:
count     30000.000000
mean       5200.366300
std       16838.019547
min           1.000000
25%          81.000000
50%         731.000000
75%        3615.250000
max      517715.000000
Name: impressions_90d, dtype: float64

Average CTR (%): 0.51


## 3. Quick Look at the Data

To understand whether this lane is worth exploring, I first examined the starter dataset.

- The dataset contains **30,000** content pages across **44** features.
- **16,262 pages (about 54.2%)** are currently labelled with a **"down"** trend, suggesting there are many potential refresh candidates.
- The **median** number of impressions over the previous 90 days is **731**, while the **mean** is **5,200.37**, indicating that page visibility is highly skewed, with a relatively small number of pages receiving very high impressions.
- The average CTR is **0.51%**, showing that there may be opportunities to improve click performance for visible pages.

These observations suggest that the dataset contains sufficient variation in search performance to support a ranking-based approach for prioritizing content review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful Words

This notebook frames a decision-support problem rather than attempting to prove cause and effect. The analysis is based on historical search and engagement signals, so it can identify patterns and prioritization opportunities but cannot prove that refreshing a page will directly increase traffic or improve rankings. Any recommendations produced in later stages should be reviewed by a human before action is taken.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x ] Every section above is filled — markdown thinking AND the code that backs it
- [x ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x ] No client names, URLs, or private queries anywhere
- [x ] My claims use careful words: observed, measured, directional, decision-support
- [ x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.